src/application/listeners/projectEventListeners.ts

```diff
@@ -1,81 +1,99 @@
     } else {
       const newProjectPath = await save({ defaultPath: await getNewProjectsBaseDir() });
-      if (typeof newProjectPath === 'string') {
-        await newProject(newProjectPath);
-        createFile();
-        persistProjectDirInSettings(newProjectPath);
-        notifyInfo('New project created at ' + newProjectPath);
+      if (typeof newProjectPath !== 'string') {
+        notifyInfo('User canceled operation to create new project');
+        return;
       }
+      const projectName = newProjectPath.split('/').pop() ?? 'Project';
+      projectInfo = await createRemoteProject(projectName, newProjectPath);
     }
+
+    await newProject(projectInfo.id, projectInfo.path, projectInfo.name);
+    createFile();
+    notifyInfo('New project created with success');
+    persistProjectIdInSettings(projectInfo.id);
   };
 }
 
 export function useFileProjectNewSampleListener() {
   const sampleProject = useSetAtom(newSampleProjectAtom);
+  const createFile = useSetAtom(createFileAtom);
   return async () => {
-    const newProjectPath = `${await getNewProjectsBaseDir()}RefStudio Sample`;
-    await sampleProject(newProjectPath);
-    persistProjectDirInSettings(newProjectPath);
-    notifyInfo('New project created at ' + newProjectPath);
+    const projects = await readAllProjects();
+    let project = projects.find((p) => p.name === SAMPLE_PROJECT_NAME);
+    if (!project) {
+      project = await createRemoteProject('RefStudio Sample');
+    }
+    await sampleProject(project.id, project.name, project.path);
+    createFile();
+    notifyInfo('Sample project opened with success');
+    persistProjectIdInSettings(project.id);
   };
 }
 
 export function useFileProjectOpenListener() {
   const openProject = useSetAtom(openProjectAtom);
-  const openWebProject = useSetAtom(openWebProjectAtom);
 
   return async () => {
+    let projectId: string;
     if (import.meta.env.VITE_IS_WEB) {
       // TODO: Open a custom dialog with all projects to select one
       // For now, opens the first project found
-      const projectInfo = await readOneProjectFromWeb();
-      return openWebProject(projectInfo.id, projectInfo.path, projectInfo.name);
-    }
-
-    const selectedPath = await open({
-      directory: true,
-      multiple: false,
-      defaultPath: await getNewProjectsBaseDir(),
-      title: 'Open RefStudio project',
-    });
+      const projects = await readAllProjects();
+      projectId = projects[0].id;
+    } else {
+      const selectedPath = await open({
+        directory: true,
+        multiple: false,
+        defaultPath: await getNewProjectsBaseDir(),
+        title: 'Open RefStudio project',
+      });
 
-    if (typeof selectedPath === 'string') {
-      await openProject(selectedPath);
-      persistProjectDirInSettings(selectedPath);
-      notifyInfo('New project open at ' + selectedPath);
+      if (typeof selectedPath === 'string') {
+        const projects = await readAllProjects();
```

Comment:
I don't feel too strongly either way, but this logic could also go in the backend (like a projectIdForPath endpoint).

Reply:
I see your point, but I would rather implement a filter by path in the existing API /projects. For now I think it's fine how it is.